src/main/java/com/code_intelligence/jazzer/junit/JUnitLifecycleMethodsInvoker.java

```diff
@@ -37,64 +45,202 @@
                                 callback.invokeAfterEachMethod(
-                                    extensionContext, extensionRegistry)))
+                                    updatingExtensionContext, extensionRegistry)))
             .collect(toCollection(ArrayList::new));
     // JUnit calls AfterEach methods in reverse order of registration so that the methods registered
     // first run last.
     Collections.reverse(afterEachMethods);
 
+    Constructor<?> constructor = getTestClassNoArgsConstructor(updatingExtensionContext);
+    ThrowingConsumer[] instancePostProcessors =
+        extensionRegistry.stream(TestInstancePostProcessor.class)
+            .map(
+                processor ->
+                    (ThrowingConsumer)
+                        (instance ->
+                            processor.postProcessTestInstance(instance, emptyExtensionContext)))
+            .toArray(ThrowingConsumer[]::new);
+    ThrowingRunnable updateTestClassInstance =
+        () -> {
+          Object instance = constructor.newInstance();
+          for (ThrowingConsumer instancePostProcessor : instancePostProcessors) {
+            instancePostProcessor.accept(instance);
+          }
+          mutableTestClassInstance[0] = instance;
+        };
+
     return new JUnitLifecycleMethodsInvoker(
-        Stream.concat(afterEachMethods.stream(), beforeEachMethods.stream())
-            .toArray(ThrowingRunnable[]::new));
+        updateTestClassInstance,
+        () -> mutableTestClassInstance[0],
+        beforeEachMethods,
+        afterEachMethods.toArray(new ThrowingRunnable[0]));
+  }
+
+  private static TestInstances makeTestInstances(
+      Class<?> clazz, Supplier<Object> singleTestInstance) {
+    return new TestInstances() {
+      @Override
+      public Object getInnermostInstance() {
+        return singleTestInstance.get();
+      }
+
+      @Override
+      public List<Object> getEnclosingInstances() {
+        return emptyList();
+      }
+
+      @Override
+      public List<Object> getAllInstances() {
+        return singletonList(singleTestInstance.get());
+      }
+
+      @Override
+      public <T> Optional<T> findInstance(Class<T> aClass) {
+        if (clazz == aClass) {
+          return (Optional<T>) Optional.of(singleTestInstance.get());
+        } else {
+          return Optional.empty();
+        }
+      }
+    };
+  }
+
+  private static Constructor<?> getTestClassNoArgsConstructor(ExtensionContext extensionContext) {
+    Class<?> testClass = extensionContext.getRequiredTestClass();
+    if (testClass.getEnclosingClass() != null) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The test class %s is an inner class, which is not supported with"
+                  + " LifecycleMode.PER_EXECUTION. Either make it a top-level class or set"
+                  + " lifecycle = LifecycleMode.PER_TEST on @FuzzTest.",
+              testClass.getName()));
+    }
+    Constructor<?> constructor;
```

Comment:
`constructor` could be looked up and returned completely in the try-catch-block, which might simplify the code a little bit.